### PR TITLE
feat: precommit hooks for docstrings

### PR DESCRIPTION
**Pull request by cristianmtr**
- [x] Add pre-commit hook for linting docstrings
- [x] Use agreed-upon configuration
- [x] Update CONTRIBUTING.md
- [x] Updated docs on docstrings https://github.com/jina-ai/docs/pull/83

### CODE REVIEW

**Comment by hanxiao** [comment]
is this mandatory, i.e. if not using `pre-commit` will it fail/block on the PR?

**Comment by maximilianwerk** [comment]
> is this mandatory, i.e. if not using `pre-commit` will it fail/block on the PR?

No, it is not. `pre-commit` will only be executed, when you have installed it consciously. I would still recommend all core developers to use it, to not get diverging quality in the checked parts.

`pre-commit` is mostly for way faster feedback to the developer, when you don't want to push and wait for CI to tell you, what you did wrong.

**Review by maximilianwerk — APPROVED** [review]
LGTM

**Pull request merged**
